### PR TITLE
Removing Propagator from Options

### LINKED ISSUE #1439: Instrumentations and overriding Propagator
https://github.com/open-telemetry/opentelemetry-dotnet/issues/581 is resolved as we added Global Propagators API which allow one to retrieve the Global propagator, and SDK sets the default to Composite (Baggae, TraceContext).

Opening this issue to discuss if we need to still allow Instrumentations to override their Propagator. Its not really mentioned in the spec, but there could be use cases where a specific instrumentation requires non-default propagator.

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #1439.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNet/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
   ([#1427](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1427))
 * Propagators.DefaultTextMapPropagator will be used as the default Propagator
   ([#1427](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1428))
+* Removed Propagator from Options
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNet/CHANGELOG.md:11]: Removed Propagator from Instrumentation Options. Instrumentation now always respect the Propagator.DefaultTextMapPropagator.

Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.AspNet/CHANGELOG.md]: just updated!

**Review by cijothomas — APPROVED** [review]

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs [hunk 2]
```diff
@@ -250,6 +236,11 @@ public async Task CustomPropagator()
 
             Assert.Equal(expectedTraceId, activity.Context.TraceId);
             Assert.Equal(expectedSpanId, activity.ParentSpanId);
+            Sdk.SetDefaultTextMapPropagator(new CompositeTextMapPropagator(new TextMapPropagator[]
```
Comment (cijothomas) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:239]: do this inside Dispose or in Ctor

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.netcore31.cs [hunk 3]
```diff
@@ -187,6 +184,11 @@ public async Task HttpClientInstrumentationInjectsHeadersAsync_CustomFormat(bool
 
             Assert.Equal($"00/{activity.Context.TraceId}/{activity.Context.SpanId}/01", traceparents.Single());
             Assert.Equal("k1=v1,k2=v2", tracestates.Single());
+            Sdk.SetDefaultTextMapPropagator(new CompositeTextMapPropagator(new TextMapPropagator[]
```
Comment (cijothomas) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.netcore31.cs:187]: move this to ctor / dispose

Comment (eddynaka) [reply test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.netcore31.cs]: moved to dispose.

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs (hunk 2 above)
Comment (eddynaka) [reply test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs]: moved to dispose.

**Pull request merged**